### DISCUSSION TITLE
Moderation doesn't work

### DISCUSSION THREAD

**Discussion by remixtedi** [betalgo/openai#512] [Q&A]
**Describe the bug**
I'm trying to check the prompt for inappropriate language before it goes to the ChatCompletions endpoint but it returns  "404 Resource not found" Error.

I'm using OpenAI services on Azure and URL looks like this:
 https://MYOPENAISERVICE.openai.azure.com/openai/deployments/Gpt4-model-0125-Deployment/moderations?api-version=2023-12-01-preview

**Your code piece**
```
 var moderationResponse = await AzureOpenAIService.CreateModeration(createModerationRequest);
        if (!moderationResponse.Successful)
            throw new ValidationException(moderationResponse.Error.Message ??
                                          "Input text is not appropriate for the audience because it might contain hate, sexual, violence, and self-harm content, please try again with different input.");
```

**Comment by kayhantolga** [comment]
I believe the problem lies in your configuration, so I'm converting this to a discussion rather than an issue.

Could you share the initialization part of your code? (Please ensure you do not share any sensitive information like keys, etc.)
